### PR TITLE
feat(sdk-coin-trx): batch consolidate native TRX to base

### PR DESCRIPTION
**Pull request by theMobiusStrip**
EA-1328

### CODE REVIEW

File: modules/sdk-coin-trx/src/trx.ts [hunk 1]
```diff
@@ -94,6 +95,18 @@ export interface RecoveryOptions {
   scan?: number;
 }
 
+export interface ConsolidationRecoveryOptions {
+  userKey: string;
+  backupKey: string;
+  bitgoKey: string;
+  startingScanIndex?: number; // default to 1 (inclusive)
+  endingScanIndex?: number; // default to startingScanIndex + 20 (exclusive)
```
Comment (mohammadalfaiyazbitgo) [inline modules/sdk-coin-trx/src/trx.ts:103]: can this be added to the javadocs for the method as well for better visibility?

File: modules/sdk-coin-trx/src/trx.ts [hunk 2]
```diff
@@ -743,6 +756,67 @@ export class Trx extends BaseCoin {
...
+  /**
+   * Builds native TRX recoveries of receive addresses in batch without BitGo.
+   * Funds will be recovered to base address first. You need to initiate another sweep txn after that.
+   * Note: there will be another recoverTokenConsolidates function to support token recover from receive addresses.
+   *
+   * @param params ConsolidationRecoveryOptions: startingScanIndex is inclusive, endingScanIndex is exclusive.
+   */
+  async recoverConsolidations(params: ConsolidationRecoveryOptions): Promise<ConsolidationRecoveryBatch> {
+    const isUnsignedConsolidations = getIsUnsignedSweep(params);
+    const startIdx = params.startingScanIndex || 1;
```
Comment (alebusse) [inline modules/sdk-coin-trx/src/trx.ts:768]: should default to 0 if you gonna use it as part of the path

**Review by kxl4126 — APPROVED** [review]
lgtm

File: modules/sdk-coin-trx/src/trx.ts [hunk 3]
```diff
@@ -707,7 +720,7 @@ export class Trx extends BaseCoin {
 
     // a sweep potentially needs to pay for multi-sig transfer, destination account activation and bandwidth
     // TRON foundation recommends 2.1 TRX for guaranteed confirmation
-    if (!recoveryAmount || SAFE_TRON_TRANSACTION_FEE > recoveryAmount) {
+    if (!recoveryAmount || SAFE_TRON_TRANSACTION_FEE >= recoveryAmount) {
```
Comment (alebusse) [inline modules/sdk-coin-trx/src/trx.ts:723]: Looks like it will always leave 1 unit behind

Comment (evanzbitgo) [reply modules/sdk-coin-trx/src/trx.ts:723]: Yes, that's how Tron works with us currently. If balance is the reserved fee, then remaining is 0, so nothing to sweep

**Review by mkottaichamy — APPROVED** [review]
Approved.

**Pull request merged**